internal/allocator.h

```diff
@@ -192,7 +192,7 @@ class Allocator {
  private:
   void DeallocateStorage() {
     assert(!committed_);
-    free(storage_);
+    aligned_free(storage_);
```

Comment:
There is a discrepancy here: platform.h defines posix_memalign so that the code here thinks it's just doing a plain old posix_memalign on all platforms, but then the contract of posix_memalign should be actually honored, and that contract includes the fact that the memory allocated by posix_memalign may be freed by free().

Of course, you can't fulfill that here, because free() is already defined on Windows, meaning something else.

The conclusion should be that you can't have posix_memalign as a polyfill'd function here.

You could rename posix_memalign into aligned_alloc so that all users can do on all platforms: aligned_alloc, aligned_free.